src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
+const FULL_HEALTH = 100;
+const DAMAGE = 50;
+
 class Animal {
-  // write your code here
+  constructor(name = '') {
+    this.health = FULL_HEALTH;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name = '') {
+    super(name);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name = '') {
+    super(name);
+  }
+  bite(production) {
+    if (production instanceof Herbivore && !production.hidden) {
+      production.health -= DAMAGE;
+    }
+
+    if (production.health === 0) {
```

Comment:
If the `health` value is less than zero, then the condition will not be fulfilled and there will be an object with a negative health value in the array of living animals